public/route_service.ts

```diff
@@ -33,80 +34,124 @@ import {
  * Used in redux by wrapping them in async thunk functions which mutate redux state when executed.
  */
 export interface RouteService {
-  getWorkflow: (workflowId: string) => Promise<any | HttpFetchError>;
-  searchWorkflows: (body: {}) => Promise<any | HttpFetchError>;
-  getWorkflowState: (workflowId: string) => Promise<any | HttpFetchError>;
-  createWorkflow: (body: {}) => Promise<any | HttpFetchError>;
+  getWorkflow: (
+    workflowId: string,
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
+  searchWorkflows: (
+    body: {},
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
+  getWorkflowState: (
+    workflowId: string,
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
+  createWorkflow: (
+    body: {},
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
   updateWorkflow: (
     workflowId: string,
     workflowTemplate: WorkflowTemplate,
-    updateFields: boolean
+    updateFields: boolean,
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
+  provisionWorkflow: (
+    workflowId: string,
+    dataSourceId?: string
   ) => Promise<any | HttpFetchError>;
-  provisionWorkflow: (workflowId: string) => Promise<any | HttpFetchError>;
   deprovisionWorkflow: (
     workflowId: string,
+    dataSourceId?: string,
     resourceIds?: string
   ) => Promise<any | HttpFetchError>;
-  deleteWorkflow: (workflowId: string) => Promise<any | HttpFetchError>;
-  getWorkflowPresets: () => Promise<any | HttpFetchError>;
-  catIndices: (pattern: string) => Promise<any | HttpFetchError>;
+  deleteWorkflow: (
+    workflowId: string,
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
+  getWorkflowPresets: (dataSourceId?: string) => Promise<any | HttpFetchError>;
+  catIndices: (
+    pattern: string,
+    dataSourceId?: string
+  ) => Promise<any | HttpFetchError>;
   searchIndex: (
     index: string,
     body: {},
+    dataSourceId?: string,
     searchPipeline?: string
   ) => Promise<any | HttpFetchError>;
```

Comment:
Having multiple optional fields that are not defined under some interface can lead to issues. For example, calling searchIndex with a valid search pipeline, but no data source ID - e.g., `searchIndex('abc', {query: match_all: {}}, 'my-search-pipeline'` would treat `my-search-pipeline` as the `dataSourceId` param and fail.

For these cases, let's wrap all of the params in an interfaced obj - something like:
```
 searchIndex: (
{
    index: string,
    body: {},
    dataSourceId?: string,
    searchPipeline?: string
  } )
```